### PR TITLE
[EBPF] Ensure user can download build image

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Ensure that the user is logged in for the ECR docker registry before pulling the image.

### CODE REVIEW

File: tasks/kernel_matrix_testing/compiler.py [hunk 1]
```diff
@@ -76,6 +76,12 @@ def start(self) -> None:
         if self.is_running:
             self.stop()
 
+        info("[+] Updating build image, ensuring we can log in to the registry")
+        self.ctx.run(
+            "aws-vault exec sso-build-stable-developer -- aws ecr --region us-east-1 get-login-password | docker login --username AWS --password-stdin 486234852809.dkr.ecr.us-east-1.amazonaws.com"
+        )
+        self.ctx.run(f"docker pull {self.image}")
```
Comment (brycekahle) [inline tasks/kernel_matrix_testing/compiler.py:83]: Do you think it makes sense to try pulling first, and then logging in if that fails? Or maybe checking if the image exists locally before logging in and trying to pull it?

Comment (gjulianm) [reply tasks/kernel_matrix_testing/compiler.py:83]: Maybe checking if the image exists is the best option, as for a given tag the image shouldn't change. Logging in only after a failure is a worse option ebcause you can't really know why the pull failed reliably.

File: tasks/kernel_matrix_testing/compiler.py [hunk 2]
```diff
@@ -76,11 +76,15 @@ def start(self) -> None:
...
             self.stop()
 
-        info("[+] Updating build image, ensuring we can log in to the registry")
-        self.ctx.run(
-            "aws-vault exec sso-build-stable-developer -- aws ecr --region us-east-1 get-login-password | docker login --username AWS --password-stdin 486234852809.dkr.ecr.us-east-1.amazonaws.com"
-        )
-        self.ctx.run(f"docker pull {self.image}")
+        # docker images command does not fail if the image is not found, so we need to grep the output
+        # to get an error exit code
+        res = self.ctx.run(f"docker images {self.image} | grep {self.image}", hide=True, warn=True)
```
Comment (brycekahle) [inline tasks/kernel_matrix_testing/compiler.py:81]: I think you can do `docker image inspect {self.image}` and just check the exit code.

Comment (gjulianm) [reply tasks/kernel_matrix_testing/compiler.py]: Done!

**Review by brycekahle — APPROVED** [review]

**Pull request merged**